src/basic/os-util.c

```diff
@@ -230,9 +230,15 @@ int open_extension_release_at(
                         continue;
                 }
 
-                if (!relax_extension_release_check &&
-                    extension_release_strict_xattr_value(fd, dir_path, de->d_name) != 0)
-                        continue;
+                if (!relax_extension_release_check) {
+                        const char *underscore = strchr(extension, '_');
+
+                        /* If we have extension_release.app and app_1.0.raw, it is a valid combination and
+                         * we allow it, otherwise check for the xattr. */
+                        if (!(underscore && strneq(image_name, extension, underscore - extension)) &&
+                            extension_release_strict_xattr_value(fd, dir_path, image_name) != 0)
+                                continue;
+                }
```

Comment:
hmm, this seems a bit ad-hoc to me...

In the new vpick stuff we derive the basename for an imageto pick from either the parent dir basename or an explicitly specified pattern, and I am pretty sure that's what we should take into consideration here too.

Or with other words: I pretty sure we should not try to separate basename and version here down in the depth of open_extension_release_at(). Instead we should probably ignore the concept of versioning as we always did – except if the image we are dealing with was determined via vpick. And in that case we should use the basename that vpick was using it.

or in other words:

consider: 

1. `/var/lib/confext/foobar_0.4.raw`
2. `/var/lib/confext/foobar.raw.v/foobar_0.4.raw`
3. `/var/lib/confext/foobar.raw`

I think in cases 2 and 3 we should use an extension basename for the check of "foobar". But in case one we should use "foobar_0.4", as before. 

This requires some minor changes to the vpick stuff: PickResult has to return in a new field the actual basename used for th version picking. And this has then to be passed along with the image name as the actual image name.

Reply:
> This same approach is taken to accept images elsewhere,

where?

Reply:
I don't follow, how is the versioning supposed to work then when multiple images are found? Isn't this simply alphanumeric ordering and thus a different behavior compared to the `.v` dir behavior?

Reply:
There is no involvement of multiple images here, this is purely about matching an extension-release file inside an image with the image name that it's contained in